src/blob/index/core.rs

```diff
@@ -296,27 +290,41 @@ where
     }
 
     async fn get_all(&self, key: &K) -> Result<Option<Vec<RecordHeader>>> {
-        match &self.inner {
+        let res = match &self.inner {
             State::InMemory(headers) => Ok(headers.get(key).cloned()),
             State::OnDisk(findex) => findex.find_by_key(key).await,
-        }
+        };
+        res.map(|r| {
+            r.and_then(|mut hs| {
+                let last_del = (hs.len() - hs.iter().rev().take_while(|h| !h.is_deleted()).count())
```

Comment:
The records are ordered from newest to latest, so you should find the first deletion record.
Also there is `position` function on `iter` that do what you need.